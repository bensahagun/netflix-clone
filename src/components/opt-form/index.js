import React from 'react';
import { Container, Input, Button, Break, Text } from './styles/opt-form';

export default function OptForm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

OptForm.Input = function ({ ...restProps }) {
  return <Input {...restProps}></Input>;
};

OptForm.Button = function ({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src='/images/icons/chevron-right.png' alt='Try now' />
    </Button>
  );
};

OptForm.Text = function ({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function ({ ...restProps }) {
  return <Break {...restProps} />;
};
