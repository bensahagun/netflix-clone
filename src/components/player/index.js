import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';
import { Container, Button, FeatureButton, Overlay, Inner, Close } from './styles/player';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);

  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} {...restProps}>
          <Inner>
            <video id='netflix-player' controls autoPlay='autoplay'>
              <source src={src} type='video/mp4' />
            </video>
          </Inner>
          <Close />
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <Button {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play
    </Button>
  );
};

Player.FeatureButton = function PlayerFeatureButton({ ...restProps }) {
  // eslint-disable-next-line no-unused-vars
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <FeatureButton {...restProps} onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play Now
    </FeatureButton>
  );
};
