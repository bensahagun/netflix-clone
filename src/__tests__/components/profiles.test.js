import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles/>', () => {
  it('renders <Profiles/> with populated data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture src='/images/ben.jpg' data-testid='profile-picture' />
            <Profiles.Name>Ben Sahagun</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId('profile-picture')).toBeTruthy();
    expect(getByText('Ben Sahagun')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders <Profiles/> with populated data but no picture given', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture data-testid='profile-picture-misc' />
            <Profiles.Name>Ben Sahagun</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId('profile-picture-misc')).toBeTruthy();
    expect(getByText('Ben Sahagun')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
