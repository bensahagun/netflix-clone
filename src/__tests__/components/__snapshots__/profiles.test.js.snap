// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles/> renders <Profiles/> with populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 fGoDNJ"
>
  <h1
    class="profiles__Title-bxtkkv-1 llHZal"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 hybA-dO"
  >
    <li
      class="profiles__User-bxtkkv-5 fbQLcA"
    >
      <img
        class="profiles__Picture-bxtkkv-4 kpLrAV"
        data-testid="profile-picture"
        src="/images/users//images/ben.jpg.png"
      />
      <p
        class="profiles__Name-bxtkkv-3 jQTvVi"
      >
        Ben Sahagun
      </p>
    </li>
  </ul>
   
</div>
`;

exports[`<Profiles/> renders <Profiles/> with populated data but no picture given 1`] = `
<div
  class="profiles__Container-bxtkkv-0 fGoDNJ"
>
  <h1
    class="profiles__Title-bxtkkv-1 llHZal"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 hybA-dO"
  >
    <li
      class="profiles__User-bxtkkv-5 fbQLcA"
    >
      <img
        class="profiles__Picture-bxtkkv-4 kpLrAV"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-bxtkkv-3 jQTvVi"
      >
        Ben Sahagun
      </p>
    </li>
  </ul>
   
</div>
`;
